File: A2/A2-019/A2-019.py
# Input string
txt = input().strip()

# Find first appearance of 'B'
firstB = txt.upper().find('B')

# Find the most consecutive 'U' in a string
idx = 0
consecutive_u = 0
max_consecutive_u = 0
while(True):
    # Find appearance of 'BUU' start at idx + 3
    idx = txt.upper().find('BUU', idx)
    # 'BUU' not found, stop the process
    if(idx == -1):
        break
    # 'BUU' found, count consecutive 'U'
    for i in range(idx + 1, len(txt)):
        if(txt[i].upper() == 'U'):
            consecutive_u += 1
        else:
            break
    # Find maximum consecutive 'U'
    max_consecutive_u = max(max_consecutive_u, consecutive_u)
    consecutive_u = 0
    # Shift idx by 3
    idx += 3

# Output (Has 'BUU')
if(max_consecutive_u > 1):
    print('Yes', max_consecutive_u)
# Output (Has no 'BUU', but has 'B')
elif(firstB != -1):
    # Output same letter until the first 'B', then spams 'U'
    print(txt[:firstB + 1] + ('U' * (len(txt) - firstB - 1)))
# Output (Has no 'BUU' nor 'B')
else:
    # Output 'BUU' in loop
    print(('BUU' * ((len(txt) // 3) + 1))[:len(txt)])

File: A2/A2-019/A2-019.cpp
#include<iostream>
using namespace std;

int firstB = -1, consecU = 0, maxConsecU = 0;
bool isCountStart = false;
string txt, newText;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input string
    cin >> txt;
    for(int i = 0; i < txt.size(); i++)
    {
        // Start counting
        if(txt[i] == 'B' || txt[i] == 'b')
        {
            // Reset counter
            isCountStart = true;
            maxConsecU = max(maxConsecU, consecU);
            consecU = 0;
            // Get index of the first appearance of 'B'
            if(firstB == -1) { firstB = i; }
        }
        else if(isCountStart && (txt[i] == 'U' || txt[i] == 'u')) 
        { 
            consecU++;
        }
        // Stop counting
        else
        {
            // Reset counter
            isCountStart = false;
            maxConsecU = max(maxConsecU, consecU);
            consecU = 0;
        }
    }
    // Find maximum consecutive 'U' once again
    maxConsecU = max(maxConsecU, consecU);
    
    // Output (Has 'BUU')
    if(maxConsecU > 1)
    {
        cout << "Yes " << maxConsecU << '\n';
    }
    // Output (Has no 'BUU', but has 'B')
    else if(firstB != -1)
    {
        // Output same letter until the first appearance of 'B'
        for(int i = 0; i <= firstB; i++)
        {
            cout << txt[i];
        }
        // Spams 'U' until the end of a text
        for(int i = firstB + 1; i < txt.size(); i++)
        {
            cout << 'U';
        }
    }
    // Output (Has no 'BUU' nor 'B')
    else
    {
        // Output 'BUU' in loop
        for(int i = 0; i < txt.size(); i++)
        {
            if(i % 3 == 0) { cout << 'B'; }
            else { cout << 'U'; }
        }
    }
    return 0;
}